### DISCUSSION TITLE
Explanation of colors on Presence page

### DISCUSSION THREAD

**Discussion by milindpatel63** [netalertx/NetAlertX#866] [General]
![image](https://github.com/user-attachments/assets/bb9ef9ce-c4c7-4c02-9cde-fd507ae824b9)

What does the blue, yellow and green color mean in the presence page?
I get green is for device is Online.
Which color is Offline? and what's the third color?

**Comment by jokob-sk** [comment]
Hi @milindpatel63 ,

No color/empty space means offline. 

Yellow is a session miss-match. Not sure why this occurs frequently. Happy to accept a PR for someone to fix it. 

Some more details can be found here:

https://github.com/jokob-sk/NetAlertX/discussions/828
https://github.com/jokob-sk/NetAlertX/issues/814

**Reply by milindpatel63** [reply]
gotcha...will look into why i'm having session miss-match.
What about Blue/Green. Does both mean the same (Online)?

**Reply by milindpatel63** [reply]
> Hi @milindpatel63 ,
> 
> No color/empty space means offline.
> 
> Yellow is a session miss-match. Not sure why this occurs frequently. Happy to accept a PR for someone to fix it.
> 
> Some more details can be found here:
> 
> #828 #814

In my case, i think those session miss-match might be due to using multiple device scanners.
I was using Arp scan, Nmap device discovery and Dnsmasq leases file (from openwrt router) all at the same time previously.
Now i disabled dnsmasq leases file.
If still the chart shows yellow after a while, i'll try just using arp scan.

**Reply by milindpatel63** [reply]
> gotcha...will look into why i'm having session miss-match. What about Blue/Green. Does both mean the same (Online)?

I figured out the Blue/Green color

Green denotes the current online session of the device.
Blue is all the previous online sessions.
Yellow is session miss-match i guess.
Blank space is offline.

**Reply by jokob-sk** [reply]
Thanks! added a small explanation for colors on that page:

![image](https://github.com/user-attachments/assets/2b61b02d-085d-48ef-9a0c-b1fe835546f9)
